linked_list/main.py

```diff
@@ -43,53 +43,173 @@ class Node(object):
+        # we can call count() to count up the nodes.
+        return self.count()
+
+    def __iter__(self): # jon:DONE
+        # start at head.  yield item, increment to next node in list.
+        current = self.start
+        while current:
+            yield current
+            current = current.next
+
+    def __getitem__(self, index): # jon:DONE
+        # fetch item at position index
+        # this is what returns self[key]
+        # TODO: BOUNDS CHECKING!  RAISE OUT OF INDEX ERRORS!
+        current = self.start
+        count = 0
+        for count in range(index):
+            current = current.next
+
+        return current
+
+    def __add__(self, other): # akshith: Done
+        res = []
+        for item in self:
+            res.append(item.elem)
+        for item in other:
+            res.append(item.elem)
+        return LinkedList(res)
+
+    def __iadd__(self, other): # akshith: Done
+        return self.__add__(other)
+
+    def __eq__(self, other): # akshith: IN PROGRESS
+        if len(self) != len(other):
+            return False
+        else:
+            return all( self[i] == other[i] for i in range(len(self)) )
+ 
+            
+
+    def append(self, elem): # jon:DONE
+        # Append a new node to the end of the list
+        newnode = Node(elem)
+        if self.end:
+            self.end.next = newnode
+        else:
+            self.start = newnode
+        # remember to update the 'end' pointer to the new last item.
+        self.end = newnode
+
+    def count(self): # jon:DONE
+        current = self.start
+        count = 0
+        while current:
+            count += 1
+            current = current.next
+        return count
+
+
+    def pop(self, index=None): # corey:IN PROGRESS
+        curr_node = self.start
+        if index == None:
+            index = len(self) - 1
+            
+        if len(self) in (0, None) or index >= len(self):
+            raise IndexError
+        elif index==None or index==len(self):
+            orig_end = self.end.elem
+            new_end = curr_node
+            while curr_node != self.end:
+                new_end = curr_node
+                curr_node = curr_node.next
+            if self.start == self.end:
+                self.start = None
+            new_end.next = None
+            self.end = new_end
+            return orig_end
+        elif index==0:
+            orig_start = self.start
+            self.start = self.start.next
+            orig_start.next = None
+            return orig_start.elem
+        else:
+            next_node = self.start.next
+            prev_node = self.start
+            for i in range(index):
+                prev_node = curr_node
+                curr_node = prev_node.next
+                next_node = curr_node.next
+            curr_node.next = None
+            our_node = curr_node.elem
+            curr_node = ''
+            prev_node.next = next_node
+            return our_node
```

Comment:
Really good work with `pop`. It's the hardest method to implement. You could optimize it a little bit, but the general idea is perfectly fine.